### PR TITLE
Implement first api call

### PR DESCRIPTION
**Pull request by sagersmith8**
### Changes:
- Added tons of readme information
- Setup thrift service
- Create client to test service
- Setup unit tests
- Fix gradle build

### Note:
- Travis CI is expected to fail until the service is published on maven central

Please Review: @Kuroshii

### CODE REVIEW

File: src/main/java/com/vite/service/ViteClient.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+import javax.annotation.PreDestroy;
+
+/**
+ * Simple client for proof of concept.
+ * TODO define interface
+ * TODO break out into own repo
+ * TODO use secure client
+ */
+public class ViteClient {
```
Comment (sagersmith8) [inline src/main/java/com/vite/service/ViteClient.java:19]: This class should probably live in its own repo eventually

Comment (sagersmith8) [reply src/main/java/com/vite/service/ViteClient.java:19]: The cli can be defined using this under the hood

File: src/main/java/com/vite/service/ViteServer.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+public class ViteServer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ViteHandler.class);
+
+    @Autowired
+    public ViteServer() {
+        new Thread(() -> {
+            LOGGER.debug("Creating Vite Server...");
+            TServerTransport serverTransport;
+            try {
+                serverTransport = new TServerSocket(9090);
```
Comment (Kuroshii) [inline src/main/java/com/vite/service/ViteServer.java:27]: Strange port number.
Edit: does that match the mappings we use?

File: src/test/java/com/vite/service/ViteHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+public class ViteHandlerTest {
+    /**
+     * Tests the {@link ViteHandler#ping()} function
+     * TODO add assertion about log statement, or better yet make ping return a String
+     */
+    @Test
+    public void testPing() {
+        ViteHandler viteService = new ViteHandler();
+        viteService.ping();
```
Comment (Kuroshii) [inline src/test/java/com/vite/service/ViteHandlerTest.java:14]: Test without asserts?

Maybe do something like?
```
try {
    //do important stuff
} catch(ExpectedException1 | ExpectedException2 e) {
     Assert.fail("Wow that shouldn't have happened");
}
```

File: src/main/java/com/vite/service/ViteClient.java (hunk 1 above)
Comment (Kuroshii) [reply src/main/java/com/vite/service/ViteClient.java:19]: If not in it's own repo, in a subbuild here so it's published seperately

File: src/main/java/com/vite/Application.java [hunk 4]
```diff
@@ -1,19 +1,21 @@
...
 @EnableWebMvc
-@ComponentScan({"com.vite.controllers"})
+@ComponentScan({"com.vite.service"})
 @SpringBootApplication
 public class Application {
     static final Logger LOGGER = Logger.getLogger(Application.class);
 
     public static void main(String[] args) {
         SpringApplication.run(Application.class, args);
+        new ViteClient();
```
Comment (Kuroshii) [inline src/main/java/com/vite/Application.java]: Bah humbug

File: src/main/java/com/vite/service/ViteClient.java (hunk 1 above)
Comment (sagersmith8) [reply src/main/java/com/vite/service/ViteClient.java:19]: We don't publish this currently, and this is going to be in its own repo, so I am not going to figure that out to just undo it

File: src/main/java/com/vite/service/ViteServer.java (hunk 2 above)
Comment (sagersmith8) [reply src/main/java/com/vite/service/ViteServer.java:27]: Via docker, no. Follow up

File: src/test/java/com/vite/service/ViteHandlerTest.java (hunk 3 above)
Comment (sagersmith8) [reply src/test/java/com/vite/service/ViteHandlerTest.java:14]: Follow up

File: src/main/java/com/vite/Application.java (hunk 4 above)
Comment (sagersmith8) [reply src/main/java/com/vite/Application.java]: Its technically not a bean.. just saying

File: src/main/java/com/vite/service/ViteClient.java (hunk 1 above)
Comment (Kuroshii) [reply src/main/java/com/vite/service/ViteClient.java:19]: I wasn't saying to figure it out, just trying to contribute to the discussion you were having with yourself. 😕

File: src/main/java/com/vite/Application.java (hunk 4 above)
Comment (Kuroshii) [reply src/main/java/com/vite/Application.java]: What I mean: Why are you initializing a client here? If the test works, then the test works.

File: src/test/java/com/vite/service/ViteHandlerTest.java (hunk 3 above)
Comment (Kuroshii) [reply src/test/java/com/vite/service/ViteHandlerTest.java:14]: What do you mean by "Follow up"? Is that an imperative (if so, what are you looking for? I can't push to your fork) or is that short for "see the follow up" (in which case, I don't see any)?

File: src/main/java/com/vite/service/ViteClient.java (hunk 1 above)
Comment (sagersmith8) [reply src/main/java/com/vite/service/ViteClient.java:19]: HAHA not mad sorry that sounds mad ha

File: src/main/java/com/vite/Application.java (hunk 4 above)
Comment (sagersmith8) [reply src/main/java/com/vite/Application.java]: Good point

**Comment by sagersmith8** [comment]
Follow ups made

**Pull request merged**